File: problem-solving/programmers/level01/문자열_내_마음대로_정렬하기.py
# https://programmers.co.kr/learn/courses/30/lessons/12915

def solution(strings, n):
    return sorted(strings, key=lambda x:(x[n], x))

# if __name__ == "__main__":
#     print(solution(["sun", "bed", "car"], 1))
#     print(solution(["abce", "abcd", "cdx"], 2))

File: problem-solving/programmers/level01/문자열_내_마음대로_정렬하기.java
// https://programmers.co.kr/learn/courses/30/lessons/12915

import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        String[] answer = {};
        ArrayList<String> arr = new ArrayList<>();
        for (int i=0; i<strings.length; i++) {
            arr.add(""+ strings[i].charAt(n) + strings[i]);
        }
        Collections.sort(arr);
        answer = new String[arr.size()];
        for (int i=0; i<arr.size(); i++) {
            answer[i] = arr.get(i).substring(1, arr.get(i).length());
        }
        return answer;
    }
}